### ISSUE TITLE
Enroll Now Start

### ISSUE THREAD

**Issue by Nayuken** [searchdiscovery/VPAL#14]
# [Enroll Now Start](https://github.com/searchdiscovery/VPAL/blob/main/Data%20Layer%20Events/Enroll%20Now%20Start.md)

CRP status - this is a new event designed for the CRP flow. Fire this event when the user starts to fill out a field in the Enroll now page. Only fire this for the user's first engagement with the form. 

### 

## Javascript Code
```js
window.dataLayer = window.dataLayer || [];
dataLayer.push({ event_data: null });  // Clear the previous event_data object.
dataLayer.push({
  "event": "enroll_now_start"
});
```
![Image](https://github.com/user-attachments/assets/27045634-76f6-4cb8-915f-522333dfd742)

**Comment by Nayuken** [comment]
Once the CSP violation is resolved, revalidation is needed to confirm data accuracy and ensure the correct tag firing structure for data sent to GA4.

testing on: https://lxp-admin--lxpdev.sandbox.my.site.com/learnercommunity/s/lxp-application-enrollment?course=2e110f48-c60c-49f4-884e-ea23ac8766e0&gtm_debug=1730388257058

## Feedback:

@rafaeltorresharvard This event currently fires whenever a user interacts with any form fields in the learner information section of the page. It should instead fire only once, triggered by the user's first interaction with any form field on the learner information page.

![Image](https://github.com/user-attachments/assets/41e5459a-5e2d-40df-80ab-18027e23a580)

# QA DataLayer Values:
```js
{
    "event": "enroll_now_start",
    "gtm.uniqueEventId": 247
}
```

**Comment by elizabethbrady-ep** [comment]
@rafaeltorresharvard this event should only fire the FIRST time a user engages with a field on the form - ensure it only fires one time ![Image](https://github.com/user-attachments/assets/5c91c227-9044-44fe-bf64-332879227cd1)

**Comment by Nayuken** [comment]
## Feedback:
@rafaeltorresharvard The expected behavior for this event is now observed whenever a user first interacts with a field on the learner information form.

![Image](https://github.com/user-attachments/assets/312b9afd-75fb-4195-b4a1-0c5605867143)

## Further's Event Validation:
| Event Parameter  | Result |
|------------------|--------|
| ep.course_name   |  Adaptive Leadership      |
| ep.program_dates |       February 24, 2025 - July 2, 2025 |
| ep.page_location/ Page Location  |    https://lxp-admin--lxpdev.sandbox.my.site.com/learnercommunity/s/lxp-application-enrollment?course=2e110f48-c60c-49f4-884e-ea23ac8766e0    |
| ep.page_title/ Page Title    |  Enroll Now \| Harvard Online      |
| ep.site_section  |      CRP  |
| ep.site_section2 |     CRP   |
| ep.type          |     CRP   |

**Comment by Nayuken** [comment]
A new issue has popped up with this event which now does not populate this event in the datalayer when a user clicks any of the fields under the learn information section of the page on https://lxp-admin--lxpdev.sandbox.my.site.com/learnercommunity/s/lxp-application-enrollment?course=2e110f48-c60c-49f4-884e-ea23ac8766e0.

![Image](https://github.com/user-attachments/assets/cd34b85d-9b5d-436d-858b-e5a3763a76fd)

**Comment by Nayuken** [comment]
@stephen-g-goss @rafaeltorresharvard

**Comment by elizabethbrady-ep** [comment]
FYI @rafaeltorresharvard @Nayuken I now see the dataLayer event passing for both courses on the SIT  site: ![Image](https://github.com/user-attachments/assets/849ebea2-73da-4328-b5f3-d0b8d0da256f)

**Comment by elizabethbrady-ep** [comment]
@Nayuken @rafaeltorresharvard I now see this event firing on both course on the SIT site, as well as the GA4 tag with additional attributes pulled from the page ![Image](https://github.com/user-attachments/assets/80581861-1f42-4eec-b456-5ecf3ba2aa71)
